impeller/renderer/backend/vulkan/yuv_conversion_vk.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_YUV_CONVERSION_VK_H_
+#define FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_YUV_CONVERSION_VK_H_
+
+#include <unordered_map>
+
+#include "impeller/base/comparable.h"
+#include "impeller/base/thread.h"
+#include "impeller/core/sampler.h"
+#include "impeller/renderer/backend/vulkan/shared_object_vk.h"
+#include "impeller/renderer/backend/vulkan/vk.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// A descriptor used to create a new YUV conversion in a conversion library.
+///
+using YUVConversionDescriptorVK =
```

Comment:
I don't know if its something we could fix by adjusting the headers, but the vk::ExternalFormatAndroid struct isn't defined for host builds which makes compiling this header a bit of a nightmare. It doesn't look like the structure chains work with forward declared types.

Reply:
Its sort of gross but I can at least get this to compile by using vk::SamplerYcbcrConversionCreateInfo instead and just assuming that the pNext contains a vk::ExternalFormatANDROID in the impl w/ ifdefs.

Reply:
Ok, it's not not-gross, but I think I made it work.